snet_cli/commands.py

```diff
@@ -504,3 +510,242 @@ def transact(self):
             return receipt, events
         else:
             self._error("Cancelled")
+
+
+class ServiceCommand(BlockchainCommand):
+
+    # We are sorting files before we add them to the .tar since an archive containing the same files in a different
+    # order will produce a different content hash; sorting order is the same default sorting order used in GNU tar.
+    def _tar_imports(self, paths, entry_path):
+        paths = sorted(paths, key=lambda unsorted_path: (len(unsorted_path.parts), str(unsorted_path.name)))
+        f = tempfile.NamedTemporaryFile()
+        tar = tarfile.open(fileobj=f, mode="w")
+        for path in paths:
+            tar.add(path, path.relative_to(entry_path))
+            self._printout("path: {}".format(str(path)))
+
+        tar.close()
+        return f
+
+    def _get_network(self):
+        if "network" in self.args and self.args.network:
+            if "eth_rpc_endpoint" in self.args and self.args.eth_rpc_endpoint:
+                network = self.args.eth_rpc_endpoint
```

Comment:
this variable should be like `eth_rpc_endpoint`, since in both cases it's an endpoint and not a network name (and `get_web3` takes an endpoint, not a network name)